docs/en/contribution/coding-guide.md

```diff
@@ -0,0 +1,128 @@
+   3. There should be 1 space between `)` and `{`
+   4. There should be 1 space between parameter splitting
+   5. If there are more than 3 chain calls, there should be a separate line for each call
+   6. If-else should be wrapped in `{}` even if the method body is only one line
+   7. etc....
+
+   Tips: **For more details, please check `checkStyle.xml` that at tools/checkstyle**
+
+5. Code segments are never repeated. If a code segment is used multiple times, you should not define it multiple times, but make it a public segment for other modules to use
+
+6. When throwing an exception, you need to throw the exception along with a hint message and the exception should be smaller in scope.Throwing overly broad exceptions promotes complex error handling code that is more likely to contain security vulnerabilities.For example, if your connector encounters an `IOException` while reading data, a reasonable approach would be to the following:
+
+   ```java
+   try {
+       // read logic
+   } catch (IOException e) {
+       throw IOException("Meet a IOException, it might has some problems between client and database", e);
+   }
+   ```
+
+7. Please check that each new code file you add contains the `Apache License Header` before submitting pr:
+
+   ```java
+   /*
+    * Licensed to the Apache Software Foundation (ASF) under one or more
+    * contributor license agreements.  See the NOTICE file distributed with
+    * this work for additional information regarding copyright ownership.
+    * The ASF licenses this file to You under the Apache License, Version 2.0
+    * (the "License"); you may not use this file except in compliance with
+    * the License.  You may obtain a copy of the License at
+    *
+    *    http://www.apache.org/licenses/LICENSE-2.0
+    *
+    * Unless required by applicable law or agreed to in writing, software
+    * distributed under the License is distributed on an "AS IS" BASIS,
+    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    * See the License for the specific language governing permissions and
+    * limitations under the License.
+    */
+   ```
+
+8. Please check the format of your code before submitting pr:
+
+   ```shell
+   ./mvnw checkstyle:check
+   ```
+
+9. Please test the entire project for compilation before submitting pr:
+
+   ```shell
+   # multi threads compile
+   ./mvnw -T 1C clean package
+   ```
+
+   ```shell
+   # single thread compile
+   ./mvnw clean package
+   ```
+
+10. Before submitting pr, please do a full unit test and integration test locally, best practice is to use the `seatunnel-examples` module's ability to self-test to ensure that the multi-engine is running properly and the results are correct
+
+11. If you submit a pr with a feature that requires updated documentation, please update the documentation
+
+12. If a new dependency is introduced in pr, you need to add the license declaration of the new dependency in the specified location
```

Comment:
The expression of license is vague, the jar dependency by connector-v2 is do not need add license.

Reply:
> The expression of license is vague, the jar dependency by connector-v2 is do not need add license.

Don't we need to declare a license when we bring in a new component? Because of the seatunnel-engine CI temporarily turns off the step of checking the dependency license, the follow-up is still to be opened.

Reply:
All connectors do not release binary packages.

You only need to reference the contents of the PR templates.

Reply:
So what should we do next? Delete or update?